onnx/test/shape_inference_test.py

```diff
@@ -288,31 +317,52 @@ def test_transpose_partial(self, _, version) -> None:
         )
 
     @parameterized.expand(all_versions_for("Transpose"))
-    def test_transpose_preexisting_incorrect_shape(self, *_) -> None:
+    def test_transpose_preexisting_incorrect_shape(self, _, version) -> None:
         graph = self._make_graph(
             [("X", TensorProto.FLOAT, (2, 3, 4))],
             [make_node("Transpose", ["X"], ["Y"], perm=[1, 0, 2])],
             [make_tensor_value_info("Y", TensorProto.FLOAT, (5, 5, 5))],
+            version,
         )
-        self.assertRaises(onnx.shape_inference.InferenceError, self._inferred, graph)
+        if version == 1:
+            self.assertRaises(RuntimeError, self._inferred, graph)
+        else:
+            # InferenceError is from https://github.com/onnx/onnx/pull/3169
+            self.assertRaises(
+                onnx.shape_inference.InferenceError, self._inferred, graph
+            )
 
     @parameterized.expand(all_versions_for("Transpose"))
-    def test_transpose_preexisting_incorrect_type(self, *_) -> None:
+    def test_transpose_preexisting_incorrect_type(self, _, version) -> None:
         graph = self._make_graph(
             [("X", TensorProto.FLOAT, (2, 3, 4))],
             [make_node("Transpose", ["X"], ["Y"], perm=[1, 0, 2])],
             [make_tensor_value_info("Y", TensorProto.STRING, (3, 2, 4))],
+            version,
         )
-        self.assertRaises(onnx.shape_inference.InferenceError, self._inferred, graph)
+        if version == 1:
+            self.assertRaises(RuntimeError, self._inferred, graph)
+        else:
+            # InferenceError is from https://github.com/onnx/onnx/pull/3169
+            self.assertRaises(
+                onnx.shape_inference.InferenceError, self._inferred, graph
+            )
 
     @parameterized.expand(all_versions_for("Transpose"))
-    def test_transpose_incorrect_repeated_perm(self, *_) -> None:
+    def test_transpose_incorrect_repeated_perm(self, _, version) -> None:
         graph = self._make_graph(
             [("X", TensorProto.FLOAT, (2, 3, 4))],
             [make_node("Transpose", ["X"], ["Y"], perm=[1, 0, 1])],
             [],
+            version,
         )
-        self.assertRaises(onnx.shape_inference.InferenceError, self._inferred, graph)
+        if version == 1:
```

Comment:
Potentially capture this logic in a function so we don’t duplicate it in many places?